Rover/sailboat.cpp

```diff
@@ -552,3 +575,113 @@ bool Sailboat::motor_assist_low_wind() const
             rover.g2.windvane.wind_speed_enabled() &&
             (rover.g2.windvane.get_true_wind_speed() < sail_windspeed_min));
 }
+
+void Sailboat::update_in_irons()
+{
+    // only enabled for separate mainsail and options bit set
+    {
+        const bool separate_mainsail_channel =
+            (channel_mainsail !=nullptr) &&
+            (channel_mainsail != rover.channel_throttle);
+
+        if (!(separate_mainsail_channel &&
+              option_is_set(OPTION::GET_OUT_OF_IRONS_IN_AUTO))) {
+            clear_in_irons();
+            return ;
+        }
+    }
```

Comment:
```suggestion
    const bool separate_mainsail_channel =
            (channel_mainsail != nullptr) &&
            (channel_mainsail != rover.channel_throttle);

    if (!(separate_mainsail_channel &&
              option_is_set(OPTION::GET_OUT_OF_IRONS_IN_AUTO))) {
        clear_in_irons();
        return ;
    }
```

Reply:
Ah, I found this pattern again further down, I guess your just trying to reduce the scope of `separate_mainsail_channel  `

Personally I think it makes the code harder to follow.

Reply:
I think the opposite. I did it this way since you objected to my used of lambda functions in a previous review of this same PR . It is fundamental good practise to limit scope of a variable to the most local scope that you can.  It helps code legibility and the optimiser.